test/udev-test.pl

```diff
@@ -1661,7 +1661,12 @@ sub run_test {
         exit($EXIT_TEST_SKIP);
 }
 
-if (!system($udev_bin, "check")) {
+if (!mkdir($udev_run)) {
+        warn "Failed to create directory $udev_run, skipping the test";
+        exit($EXIT_TEST_SKIP);
```

Comment:
Sure, I'll do that. Also have a look at #11098.

Reply:
I took a look at that PR and it seems that the commit from it is already included here.

Reply:
The whole script is pretty fragile: setup, cleanup and error handling are mixed. So I split changes into one trivial PR and one slightly more complicated.

Reply:
I agree that it's fragile (and PRs improving it are welcome) but because of that fragility #11098 can't be merged because the test will be just skipped everywhere if `test/run` isn't created.

Reply:
Good point. So I'll close that PR.

Please see if I put the directory creation code fragment in the right place.